File: src/Angular2AutoSaveCommands/angular2App/app/services/CommandDto.ts
export class CommandDto {

    public CommandType: string;
    public PayloadType: string;
    public Payload: any;
    public ActualClientRoute: string;

    constructor(commandType: string, payloadType: string, payload: any, actualClientRoute: string) {

        this.CommandType = commandType;
        this.PayloadType = payloadType;
        this.Payload = payload;
        this.ActualClientRoute = actualClientRoute;

    }
}

File: src/AngularAutoSaveCommands/Models/CommandDto.cs
using Newtonsoft.Json.Linq;

namespace AngularAutoSaveCommands.Models
{
    public class CommandDto
    {
        public string CommandType { get; set; }

        public string PayloadType { get; set; }

        public JObject Payload { get; set; }

        public string ActualClientRoute { get; set;}

    }
}
